lib/internal/Magento/Framework/App/DeploymentConfig/Reader.php

```diff
@@ -108,11 +113,14 @@ public function load($fileKey = null)
                 $configFile = $path . '/' . $this->configFilePool->getPath($fileKey);
                 if ($fileDriver->isExists($configFile)) {
                     $fileData = include $configFile;
+                    if (!is_array($fileData)) {
+                        throw new RuntimeException(new Phrase("Invalid configuration file: '%1'", [$configFile]));
+                    }
                 } else {
                     continue;
                 }
                 $allFilesData[$configFile] = $fileData;
-                if (is_array($fileData) && count($fileData) > 0) {
+                if (!empty($fileData)) {
```

Comment:
Please replace to an equivalent `if ($fileData)`. As always, please squash changes into a single commit.

Reply:
Done, done and done.